week3/homework/src/actions/markAsNotDone.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+async function markAsNotDone(todo, request, response) {
+  try {
+    let todoList = await todo.list().then(todoList => JSON.parse(todoList));
+    const index = parseInt(request.params.id);
+    if (todoList[index].done === false) {
```

Comment:
Same note about checking the existence of the item you are manipulating.

Reply:
Many thanks... Done.